frappe/utils/data.py

```diff
@@ -635,76 +637,110 @@ def format_datetime(datetime_string: DateTimeLikeObject, format_string: str | No
 	return formatted_datetime
 
 
-def format_duration(seconds, hide_days=False):
-	"""Converts the given duration value in float(seconds) to duration format
+DURATION_DATA = {
+	"years": {"factor": 12 * 1000000000, "abbr": "y"},
+	"months": {"factor": 1000000000, "abbr": "mo"},
+	"weeks": {"factor": 7 * 24 * 60 * 60, "abbr": "w"},
+	"days": {"factor": 24 * 60 * 60, "abbr": "d"},
+	"hours": {"factor": 60 * 60, "abbr": "h"},
+	"minutes": {"factor": 60, "abbr": "min"},
+	"seconds": {"factor": 1, "abbr": "s"},
+}
+
 
-	example: converts 12885 to '3h 34m 45s' where 12885 = seconds in float
+def format_duration(seconds: int, duration_options: dict[str, bool] | None = None) -> str:
 	"""
+	Converts the given duration in seconds to a formatted duration string.
+	Example: converts 12885 to '3h 34min 45s'.
+
+	Args:
+	        seconds (int): The duration in seconds.
+	        duration_options (dict, optional): Formatting options.
+	                Example: {"hide_days": True, "hide_months": True}
+	Returns:
+	        str: Formatted string with the duration broken down into all unit parts not hidden.
+	"""
+
+	# Be tolerant, allow set instead of dict, and bool for BC.
+	if isinstance(duration_options, set):
+		duration_options = {option: True for option in duration_options}
+	elif isinstance(duration_options, bool):
+		duration_options = {"hide_days": duration_options}
 
 	seconds = cint(seconds)
+	if not seconds:
+		return ""
 
-	total_duration = {
-		"days": math.floor(seconds / (3600 * 24)),
-		"hours": math.floor(seconds % (3600 * 24) / 3600),
-		"minutes": math.floor(seconds % 3600 / 60),
-		"seconds": math.floor(seconds % 60),
-	}
+	components = []
+	duration = seconds_to_duration(seconds, duration_options)
```

Comment:
Logic moved to new function seconds_to_duration(), mimicking how it has been done in JS.